src/Chunk.ts

```diff
@@ -595,57 +575,61 @@ export default class Chunk {
 		this.graph.scope.deshadow(toDeshadow, this.orderedModules.map(module => module.scope));
 	}
 
-	private getCheckReexportDeclarations(): { [id: string]: ReexportSpecifier[] } {
```

Comment:
I see you inlined this function again. Is there any particular reason for doing so as the updated code can still be extracted again quite safely?

Reply:
I found it was stopping my own ability to read and understand the code, so did it as part of the refactoring to try to understand the code better. I think there may still be a way to simplify this function further that would break this encapsulation, I just haven't quite found it yet, so I feel a bit more comfortable having these things together. If you think of this as algebra (which it basically is), I think there's a better simplification to be found in this form (as opposed to times when separating terms leads to better simplification).

Reply:
Well, we can think about this again once we revisit this code again the next time. In my experience, sometimes it is just finding the right name that aptly describes what is happening, sometimes one needs to dig deeper to "discover" some new abstract steps and concepts that lead to new powerful refactorings.

I like the idea of thinking "how would I describe what this function does to an outsider" and then try to structure the main controlling function to be as readable as that description.